libiio compile error

Maintainer: @mhei
Environment: LEDE master r5003

Description:
Unable to build libiio
```
make[2]: Entering directory '/home/stijn/Development/LEDE/openwrt-packages/libs/libiio'
touch /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/.prepared_34b468adedf3b56863cefa041dd52ef3_6664517399ebbbc92a37c5bb081b5c53_check
. /home/stijn/Development/LEDE/source/include/shell.sh; gzip -dc /home/stijn/Development/LEDE/source/dl/libiio-0.10.tar.gz | tar -C /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/.. -xf - 
[ ! -d ./src/ ] || cp -fpR ./src/* /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10
touch /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/.prepared_34b468adedf3b56863cefa041dd52ef3_6664517399ebbbc92a37c5bb081b5c53
rm -f /home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/stamp/.libiio_installed
mkdir -p /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10
(cd /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10; CFLAGS="-Os -pipe -mcpu=arm1176jzf-s -mfpu=vfp -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -iremap/home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10:libiio-0.10 -Wformat -Werror=format-security -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wl,-z,now -Wl,-z,relro " CXXFLAGS="-Os -pipe -mcpu=arm1176jzf-s -mfpu=vfp -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -iremap/home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10:libiio-0.10 -Wformat -Werror=format-security -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wl,-z,now -Wl,-z,relro " LDFLAGS="-L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/lib -znow -zrelro " cmake -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_VERSION=1 -DCMAKE_SYSTEM_PROCESSOR=arm -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_C_COMPILER="/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-gcc" -DCMAKE_C_COMPILER_ARG1="" -DCMAKE_CXX_COMPILER="/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-g++" -DCMAKE_CXX_COMPILER_ARG1="" -DCMAKE_ASM_COMPILER="/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-gcc" -DCMAKE_ASM_COMPILER_ARG1="" -DCMAKE_EXE_LINKER_FLAGS:STRING="-L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/lib -znow -zrelro" -DCMAKE_MODULE_LINKER_FLAGS:STRING="-L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/lib -znow -zrelro -Wl,-Bsymbolic-functions" -DCMAKE_SHARED_LINKER_FLAGS:STRING="-L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/lib -znow -zrelro -Wl,-Bsymbolic-functions" -DCMAKE_AR="/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-gcc-ar" -DCMAKE_NM="/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-gcc-nm" -DCMAKE_RANLIB="/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-gcc-ranlib" -DCMAKE_FIND_ROOT_PATH="/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/usr;/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi" -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=BOTH -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY -DCMAKE_STRIP=: -DCMAKE_INSTALL_PREFIX=/usr -DDL_LIBRARY=/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi -DCMAKE_PREFIX_PATH=/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi -DCMAKE_SKIP_RPATH=TRUE  /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10 )
-- The C compiler identification is GNU 5.4.0
-- Check for working C compiler: /home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-gcc
-- Check for working C compiler: /home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/bin/arm-openwrt-linux-muslgnueabi-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test HAS_WPEDANTIC
-- Performing Test HAS_WPEDANTIC - Success
-- Looking for strdup
-- Looking for strdup - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for newlocale
-- Looking for newlocale - found
-- Looking for in6addr_any
-- Looking for in6addr_any - found
-- Found Git: /home/stijn/Development/LEDE/source/staging_dir/host/bin/git (found version "2.13.6") 
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
-- Performing Test HAS_O_TMPFILE
-- Performing Test HAS_O_TMPFILE - Success
-- Performing Test WITH_NETWORK_EVENTFD
-- Performing Test WITH_NETWORK_EVENTFD - Success
-- Performing Test HAS_PIPE2
-- Performing Test HAS_PIPE2 - Success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Found ZLIB: /home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/usr/lib/libz.so (found version "1.2.11") 
-- Found PythonInterp: /home/stijn/Development/LEDE/source/staging_dir/hostpkg/bin/python (found version "2.7.14") 
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Doxygen not found, API documentation won't be generated
-- Found BISON: /home/stijn/Development/LEDE/source/staging_dir/host/bin/bison (found version "3.0.4") 
-- Found FLEX: /home/stijn/Development/LEDE/source/staging_dir/host/bin/flex (found version "2.6.4") 
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of struct usb_functionfs_descs_head_v2
-- Check size of struct usb_functionfs_descs_head_v2 - done
-- Looking for pthread_setname_np
-- Looking for pthread_setname_np - found
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_ASM_COMPILER
    CMAKE_ASM_COMPILER_ARG1
    CMAKE_CXX_COMPILER
    CMAKE_CXX_COMPILER_ARG1
    CMAKE_CXX_FLAGS_RELEASE
    CMAKE_MODULE_LINKER_FLAGS
    DL_LIBRARY


-- Build files have been written to: /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10
rm -f /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/.configured_*
touch /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/.configured_68b329da9893e34099c7d8ad5cb9c940
rm -f /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/.built
touch /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/.built_check
CFLAGS="-Os -pipe -mcpu=arm1176jzf-s -mfpu=vfp -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -iremap/home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10:libiio-0.10 -Wformat -Werror=format-security -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wl,-z,now -Wl,-z,relro  -I/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/usr/include -I/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/include -I/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/usr/include -I/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/include/fortify -I/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/include " CXXFLAGS="-Os -pipe -mcpu=arm1176jzf-s -mfpu=vfp -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -iremap/home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10:libiio-0.10 -Wformat -Werror=format-security -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wl,-z,now -Wl,-z,relro  -I/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/usr/include -I/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/include -I/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/usr/include -I/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/include/fortify -I/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/include " LDFLAGS="-L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/usr/lib -L/home/stijn/Development/LEDE/source/staging_dir/toolchain-arm_arm1176jzf-s+vfp_gcc-5.4.0_musl_eabi/lib -znow -zrelro " make  -C /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/. AR="arm-openwrt-linux-muslgnueabi-gcc-ar" AS="arm-openwrt-linux-muslgnueabi-gcc -c -Os -pipe -mcpu=arm1176jzf-s -mfpu=vfp -g3 -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -iremap/home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10:libiio-0.10 -Wformat -Werror=format-security -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wl,-z,now -Wl,-z,relro" LD=arm-openwrt-linux-muslgnueabi-ld NM="arm-openwrt-linux-muslgnueabi-gcc-nm" CC="arm-openwrt-linux-muslgnueabi-gcc" GCC="arm-openwrt-linux-muslgnueabi-gcc" CXX="arm-openwrt-linux-muslgnueabi-g++" RANLIB="arm-openwrt-linux-muslgnueabi-gcc-ranlib" STRIP=arm-openwrt-linux-muslgnueabi-strip OBJCOPY=arm-openwrt-linux-muslgnueabi-objcopy OBJDUMP=arm-openwrt-linux-muslgnueabi-objdump SIZE=arm-openwrt-linux-muslgnueabi-size CROSS="arm-openwrt-linux-muslgnueabi-" ARCH="arm" CMAKE_COMMAND='/home/stijn/Development/LEDE/source/staging_dir/host/bin/cmake' CMAKE_DISABLE_cmake_check_build_system=1 ;
make[3]: Entering directory '/home/build/lede/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10'
make[4]: Entering directory '/home/build/lede/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10'
make[5]: Entering directory '/home/build/lede/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10'
Scanning dependencies of target iio
make[5]: Leaving directory '/home/build/lede/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10'
make[5]: Entering directory '/home/build/lede/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10'
[  2%] Building C object CMakeFiles/iio.dir/backend.c.o
[  5%] Building C object CMakeFiles/iio.dir/channel.c.o
[  8%] Building C object CMakeFiles/iio.dir/device.c.o
[ 11%] Building C object CMakeFiles/iio.dir/context.c.o
[ 14%] Building C object CMakeFiles/iio.dir/buffer.c.o
[ 17%] Building C object CMakeFiles/iio.dir/utilities.c.o
[ 20%] Building C object CMakeFiles/iio.dir/scan.c.o
[ 23%] Building C object CMakeFiles/iio.dir/local.c.o
[ 26%] Building C object CMakeFiles/iio.dir/usb.c.o
[ 29%] Building C object CMakeFiles/iio.dir/network.c.o
[ 32%] Building C object CMakeFiles/iio.dir/xml.c.o
In file included from /usr/include/libxml2/libxml/parser.h:810:0,
                 from /usr/include/libxml2/libxml/globals.h:18,
                 from /usr/include/libxml2/libxml/threads.h:35,
                 from /usr/include/libxml2/libxml/xmlmemory.h:218,
                 from /usr/include/libxml2/libxml/tree.h:1307,
                 from /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/xml.c:23:
/usr/include/libxml2/libxml/encoding.h:31:26: fatal error: unicode/ucnv.h: No such file or directory
compilation terminated.
make[5]: *** [CMakeFiles/iio.dir/build.make:303: CMakeFiles/iio.dir/xml.c.o] Error 1
make[5]: Leaving directory '/home/build/lede/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10'
make[4]: *** [CMakeFiles/Makefile2:68: CMakeFiles/iio.dir/all] Error 2
make[4]: Leaving directory '/home/build/lede/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10'
make[3]: *** [Makefile:130: all] Error 2
make[3]: Leaving directory '/home/build/lede/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10'
make[2]: *** [Makefile:106: /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/.built] Error 2
make[2]: Leaving directory '/home/stijn/Development/LEDE/openwrt-packages/libs/libiio'
make[1]: *** [package/Makefile:111: package/feeds/packages/libiio/compile] Error 2
make[1]: Leaving directory '/home/stijn/Development/LEDE/source'
make: *** [/home/stijn/Development/LEDE/source/include/toplevel.mk:209: package/libiio/compile] Error 2
```

At first I thought libxml2 detected host ICU and enabled it due to that, but it seems the problem is really with libiio using host libxml2 headers rather than from staging_dir. This is clearly shown in the compile error:

```
[ 32%] Building C object CMakeFiles/iio.dir/xml.c.o
In file included from /usr/include/libxml2/libxml/parser.h:810:0,
                 from /usr/include/libxml2/libxml/globals.h:18,
                 from /usr/include/libxml2/libxml/threads.h:35,
                 from /usr/include/libxml2/libxml/xmlmemory.h:218,
                 from /usr/include/libxml2/libxml/tree.h:1307,
                 from /home/stijn/Development/LEDE/source/build_dir/target-arm_arm1176jzf-s+vfp_musl_eabi/libiio-0.10/xml.c:23:
/usr/include/libxml2/libxml/encoding.h:31:26: fatal error: unicode/ucnv.h: No such file or directory
compilation terminated.
```

--- a/libs/libxml2/Makefile
+++ b/libs/libxml2/Makefile
@@ -118,6 +118,10 @@ define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/lib
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libxml2.{la,a,so*} $(1)/usr/lib/
 
+	$(INSTALL_DIR) $(1)/usr/lib/cmake/libxml2
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/cmake/libxml2/libxml2-config.cmake \
+		$(1)/usr/lib/cmake/libxml2
+
 	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
 	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/libxml-2.0.pc $(1)/usr/lib/pkgconfig/
 

